catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/transformer/GmdTransformer.java

```diff
@@ -210,187 +306,596 @@ private void copyArgumentsToContext(MarshallingContext context,
+                }
+            }
 
-        List<String> subjects = new ArrayList<>();
-        if (CollectionUtils.isNotEmpty(keywords)) {
-            subjects.addAll(keywords);
+        } catch (ParserConfigurationException | IOException | SAXException | XPathExpressionException e) {
+            LOGGER.debug(
+                    "Unable to parse location in XML document.  Metacard location will not be set.",
+                    e);
+            return;
         }
-        if (CollectionUtils.isNotEmpty(topics)) {
-            subjects.addAll(topics);
+    }
+
+    private void setMetacardDates(Metacard metacard,
+            final XstreamPathValueTracker pathValueTracker) {
+        try (InputStream inputStream = new ByteArrayInputStream(byteArray)) {
+            DocumentBuilderFactory domFactory = DocumentBuilderFactory.newInstance();
+            domFactory.setNamespaceAware(false);
+            DocumentBuilder builder = domFactory.newDocumentBuilder();
+            Document document = builder.parse(inputStream);
+
+            XPath xPath = XPathFactory.newInstance()
+                    .newXPath();
+            XPathExpression dataExpression = xPath.compile(GmdConstants.RESOURCE_DATE_PATH);
+            NodeList dataList = (NodeList) dataExpression.evaluate(document,
+                    XPathConstants.NODESET);
+
+            if (dataList != null && dataList.getLength() > 0) {
+                List<String> dateList = new ArrayList<>();
+                List<String> dateTypes =
+                        pathValueTracker.getAllValues(toPath(GmdConstants.CITATION_DATE_TYPE_PATH));
+
+                for (int i = 0; i < dataList.getLength(); i++) {
+                    Node node = dataList.item(i);
+                    String datestring = node.getTextContent()
+                            .trim()
+                            .replaceAll("(\\s|\\t)+", " ")
+                            .replaceAll("(\\n)+", " ");
+                    String[] stringArray = datestring.split(" ");
+                    dateList.add(stringArray[0]);
+                }
+                if(CollectionUtils.isNotEmpty(dateList) && CollectionUtils.isNotEmpty(dateTypes) && dateList.size() == dateTypes.size()) {
+                    setDates(dateList, dateTypes, metacard);
+                }
+            }
+        } catch (ParserConfigurationException | IOException | SAXException | XPathExpressionException e) {
+            LOGGER.debug(
+                    "Unable to parse dates in XML document.  Metacard Created / Effective / Modified dates will not be set.",
+                    e);
+            return;
         }
+    }
 
-        if (subjects.size() > 0) {
-            metacard.setAttribute(GmdMetacardType.GMD_SUBJECT, (Serializable) subjects);
+    private void setDates(List<String> dates, List<String> dateTypes, Metacard metacard) {
+        for (int i = 0; i < dates.size(); i++) {
+            if (i < dates.size() && i < dateTypes.size()) {
```

Comment:
Just put this into the for statement instead of a separate if inside the for. Then you'll exit the loop early if this condition is unmet.